plugins/coreboot/fu-coreboot-misc.c

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2019 9elements Agency GmbH <patrick.rudolph@9elements.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <string.h>
+#include <stdio.h>
+
+#include "fu-plugin-coreboot.h"
+
+/* Tries to convert the coreboot version string to a triplet string.
+ * Returns NULL on error.
+ */
+const gchar *
+fu_plugin_coreboot_version_string_to_triplet (const gchar *coreboot_version,
+					      GError **error)
+{
+	guint cb_version = 0;
+	guint cb_major = 0;
+	guint cb_minor = 0;
+	guint cb_build = 0;
+	gint rc;
+
+	rc = sscanf (coreboot_version, "CBET%u %u.%u-%u", &cb_version, &cb_major,
+			&cb_minor, &cb_build);
+	if (rc < 0) {
```

Comment:
I think you want a warning or debug statement with dirty strings

Reply:
why? this is the essential part about coreboot version detection

Reply:
I would think that outputting the dirty strings are useful for debugging, but if you don't think so that's fine.